src/user.rs

```diff
@@ -297,28 +301,110 @@ impl User<Synching> {
         };
 
         let room_id = &rooms[room];
-        let content = AnyMessageEventContent::RoomMessage(MessageEventContent::text_plain(
+        let content = AnyMessageLikeEventContent::RoomMessage(RoomMessageEventContent::text_plain(
             get_random_string(),
         ));
         let instant = Instant::now();
-        let response = client.room_send(room_id, content, None).await;
-
-        match response {
-            Ok(response) => {
-                self.send(Event::RequestDuration((
-                    UserRequest::SendMessage,
-                    instant.elapsed(),
-                )))
-                .await;
-
-                self.send(Event::MessageSent(response.event_id.to_string()))
+        if let Some(room) = client.get_joined_room(room_id) {
```

Comment:
I would `unwrap_or_else` here, the join **must** be there already.

Reply:
That's true. I'd go for `expect` since we don't have a default room to provide to `unwrap_or_else`. Wdyt?